File: java/src/test/java/problems201_300/SlidingWindowMaximumTest.java
package problems201_300;

import java.util.Arrays;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class SlidingWindowMaximumTest extends SampledTest {
    @Test
    public void test() {
        for (SlidingWindowMaximum solution : SlidingWindowMaximum.values()) {
            for (SlidingWindowMaximumSample sample
                : getSamples(SlidingWindowMaximumSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.maxSlidingWindow(sample.input.nums, sample.input.k))
                    .asList()
                    .containsExactlyElementsIn(Arrays.stream(sample.output).boxed().toList());
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems201_300/SlidingWindowMaximumTest.kt
package problems201_300

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class SlidingWindowMaximumTest : SampledTest() {
    @Test
    fun test() =
        SlidingWindowMaximum.entries.forEach { solution ->
            getSamples<Array<SlidingWindowMaximumSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.maxSlidingWindow(input.nums, input.k))
                    .asList()
                    .containsExactlyElementsIn(output.toList())
            }
        }
}
